addon/mixins/data-adapter-mixin.js

```diff
@@ -125,7 +126,9 @@ export default Mixin.create({
   },
 
   authorize() {
-    assert('The `authorize` method should be overridden in your application adapter. It should accept a single argument, the request object.');
+    let [ major_version ] = DS.VERSION.split(".");
+
+    assert('The `authorize` method should be overridden in your application adapter. It should accept a single argument, the request object.', major_version >= 2);
```

Comment:
```suggestion
    assert('The `authorize` method should be overridden in your application adapter. It should accept a single argument, the request object.', major_version <= 2);
```

Isn't this what we want since in newer versions you should be using `headers` instead?

Reply:
The `assert` only throws the error if the condition is false, and I think we want it not to be thrown on newer versions 🤔

Reply:
🤦‍♂